py/client/pydeephaven/updateby.py

```diff
@@ -0,0 +1,239 @@
+#
+#     Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import List, Union
+from .dherror import DHError
+from pydeephaven.proto import table_pb2
+
+
+GrpcUpdateByOperation = table_pb2.UpdateByRequest.UpdateByOperation
+GrpcUpdateByColumn = GrpcUpdateByOperation.UpdateByColumn
+GrpcUpdateBySpec = GrpcUpdateByColumn.UpdateBySpec
+GrpcUpdateByEma = GrpcUpdateBySpec.UpdateByEma
+GrpcUpdateByEmaOptions = GrpcUpdateByEma.UpdateByEmaOptions
+GrpcUpdateByEmaTimescale = GrpcUpdateByEma.UpdateByEmaTimescale
+GrpcUpdateByEmaTicks = GrpcUpdateByEmaTimescale.UpdateByEmaTicks
+GrpcUpdateByEmaTime = GrpcUpdateByEmaTimescale.UpdateByEmaTime
+GrpcMathContext = table_pb2.MathContext
+
+
+class UpdateByBase(ABC):
+    @abstractmethod
+    def make_grpc_request(self):
+        ...
+
+
+class MathContext(Enum):
+    """An Enum for predefined precision and rounding settings in numeric calculation."""
+
+    UNLIMITED = 0, table_pb2.MathContext.RoundingMode.HALF_UP
+    """unlimited precision arithmetic, rounding is half-up"""
+
+    DECIMAL32 = 7, table_pb2.MathContext.RoundingMode.HALF_EVEN
+    """a precision setting matching the IEEE 754R Decimal32 format, 7 digits, rounding is half-even"""
```

Comment:
```suggestion
    """a precision setting matching the IEEE 754R Decimal32 format (32 bits, 7 decimal digits), rounding is half-even"""
```
same change on the others